src/analyses/threadFlag.ml

```diff
@@ -0,0 +1,87 @@
+(** Multi-threadedness analysis. *)
+
+module GU = Goblintutil
+module LF = LibraryFunctions
+
+open Prelude.Ana
+open Analyses
+
+let is_multi_ask (ask: Queries.ask): bool =
+  match ask Queries.SingleThreaded with
+  | `Bool b -> not b
+  | `Top -> true
+  | _ -> failwith "is_multi_ask"
+let is_multi ctx: bool = is_multi_ask ctx.ask
```

Comment:
Why do we need both of these here? Any `ctx` would also contain a `ask`, so what is the downside of using `is_multi_ask ctx.ask` vs. `is_multi ctx`? Passing around the entire `ctx` here confused me a bit at first.